### ISSUE TITLE
pep484 type annotation support at readthedocs.org

### ISSUE THREAD

**Issue by abulka** [rtfd/readthedocs.org#2919]
## Details

My python 3.6 project sphinx documentation builds ok locally but stumbles on python 3 type annotations encountered in a .. autodoc of a module containing such annotated functions.

* Project URL: http://sphinx-rich.readthedocs.io/en/latest/docstrings.html#google-style
* Build URL (if applicable): https://readthedocs.org/projects/sphinx-rich/builds/5501204/ (open the 5th collapsable build line)

* Read the Docs username (if applicable): abulka

## Expected Result

*A description of what you wanted to happen*

The google style docstring incl correct parsing of 
def function_with_pep484_type_annotations(param1: int, param2: str) -> bool:
should be in the documentation.

## Actual Result

*A description of what actually happened*

stacktrace

  File "/home/docs/checkouts/readthedocs.org/user_builds/sphinx-rich/checkouts/latest/lib/example_google.py", line 66
    def function_with_pep484_type_annotations(param1: int, param2: str) -> bool:
                                                    ^
SyntaxError: invalid syntax

and the google docstring example fails to be included in the resulting documentation.

**Comment by garciparedes** [comment]
I have the same problem

**Comment by garciparedes** [comment]
@abulka Did you find any solution for this? It works properly on local but readthedocs throws the same error as you.

**Comment by abulka** [comment]
No I didn't, sorry.  I just switched to building the docs locally for now.
My readthe docs build is still sitting in error.

**Comment by stsewd** [comment]
I was able to reproduce this problem on my local instance with a little example

```
  File "/home/stsewd/rtd/readthedocs.org/user_builds/type-hints-test/checkouts/latest/lib/example.py", line 1
    def function_with_pep484_type_annotations(param1: int, param2: str) -> bool:
                                                    ^
SyntaxError: invalid syntax
```

Then I change the interpreter to python 3.x (advanced settings on read the docs site) and it works.

![screenshot-2018-1-1 no title type_hints documentation](https://user-images.githubusercontent.com/4975310/34471272-dfa7cee6-ef12-11e7-82c3-8dcdb2e25c45.png)

Looks like type hints are only supported by python3 https://www.python.org/dev/peps/pep-0484/#other-backwards-compatible-conventions.

Note: your project will fail to build because some incompatible dependencies  that now have been updated (django for example, your project use 1.11, and 2.0 is installed). It's a good practice to pin your dependencies (django==1.11.1 on your requirements.txt) to avoid this problems.
